programs/ssl/ssl_client2.c

```diff
@@ -35,7 +35,7 @@ int main( void )
 
 /* Size of memory to be allocated for the heap, when using the library's memory
  * management and MBEDTLS_MEMORY_BUFFER_ALLOC_C is enabled. */
-#define MEMORY_HEAP_SIZE      120000
+#define MEMORY_HEAP_SIZE      180000
```

Comment:
How were the new memory values determined? Are they now guaranteed to be enough to handle all certificates?

Reply:
The commit messages have been updated with descriptive tests.
The original memory heap size was enough to handle all the original certificates. I have added a new extension (defined by the standard) that increased the sizes of a few of them. I empirically increased the heap size with 50% and it was more than enough to handle all. To optimize things it can be decreased, but this is a safe value with the current ones. Unfortunately I have no deep knowledge in this part of the library so if we want to determine another value I will need help.